src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -3371,6 +3371,11 @@ private OneOrMany<Symbol> GetSemanticSymbols(
                         {
                             if (resultKind == LookupResultKind.NotCreatable)
                             {
+                                if (expr.Type is { IsStatic: true } or { TypeKind: TypeKind.Interface or TypeKind.TypeParameter })
```

Comment:
>if (expr.Type is { IsStatic: true } or { TypeKind: TypeKind.Interface or TypeKind.TypeParameter }) [](http://example.com/codeflow?start=32&length=98)

The condition doesn't look anything like a check for "only constructor symbols", which is sated as the goal of this PR. I suggest we do that instead. I.e. check for what you are looking for rather than checking for what you are not looking for.